packages/pg-pool/README.md

```diff
@@ -331,6 +331,20 @@ var bluebirdPool = new Pool({
 
 __please note:__ in node `<=0.12.x` the pool will throw if you do not provide a promise constructor in one of the two ways mentioned above.  In node `>=4.0.0` the pool will use the native promise implementation by default; however, the two methods above still allow you to "bring your own."
 
+## error management
+
+Errors throwed by the pool are instances of `PoolError`. Each `PoolError` has a message (short description of the exception) and a structured code composed of 5 characters string starting with letter `Z`.
```

Comment:
- How about meaningful text like the error codes in the Node standard library?
- I’m not sure if most of these deserve their own code. Nobody should be handling library misuse like multiple release, multiple end, use after end, or type errors programmatically. “Timeout exceeded when trying to connect” makes sense, though.
- throwed → thrown

Reply:
> How about meaningful text like the error codes in the Node standard library?

Current implementation mimic the `NoticeOrError` interface from `pg-protocol`, cf [this message](https://github.com/brianc/node-postgres/pull/2498/files#r663694243 )

> Nobody should be handling library misuse like multiple release, multiple end, use after end, or type errors programmatically.

That's why it raises ~exceptions~ errors.

> “Timeout exceeded when trying to connect” makes sense, though.

That's actually the only one I want to catch for now. For instance in a HTTP API, my goal is to return (a more) meaningful message to the client (in this case a `503` which means that the service is unavailable for now).

Reply:
I don’t think mimicking the format of PostgreSQL error codes (with values that don’t come from PostgreSQL) is useful. https://nodejs.org/api/errors.html#nodejs-error-codes